core/src/test/scala/org/http4s/rho/AuthedContextSpec.scala

```diff
@@ -26,13 +26,22 @@ object MyAuth extends AuthedContext[IO, User]
 object MyService extends RhoService[IO] {
   import MyAuth._
 
-  GET +? param("foo", "bar") >>> auth |>> { (req: Request[IO], foo: String, user: User) =>
+  GET +? param("foo", "bar") >>> auth |>> { (_: Request[IO], foo: String, user: User) =>
     if (user.name == "Test User") {
       Ok(s"just root with parameter 'foo=$foo'")
     } else {
       BadRequest("This should not have happened.")
     }
   }
+
+  GET / "public" / 'place |>> { path: String => Ok(s"not authenticated at $path") }
+
+  GET / "private" / 'place |>> { (req: Request[IO], path: String) =>
+    getAuth(req) match {
+      case Some(user) => Ok(s"${user.name} at $path")
+      case None => Forbidden(s"not authenticated at $path")
```

Comment:
Unfortunately, if `getAuth` returns `None` it doesn't mean the user was not authenticated, it means the `AuthMiddleware` wasn't applied.

Reply:
Our `authenticated` middleware always returns `Some`, but in case where it returns `None` (which is still valid) - it will be indistinguishable from not applied middleware. So I have we have plenty of slightly broken APIs, but this one from my POV is less broken. We just need to mention in comments that `AuthMiddleware` always should be applied. Until we come up with best strategy.